### PR TITLE
Bug 1800700 - Add default search suggest header

### PR DESCRIPTION
**Pull request by sarah541**
| | |
|----|----|
|Screenshot_20221129-141015|Screenshot_20221129-141038|

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [x] **QA Needed**

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-debug` task.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation

Used by GitHub Actions.

### CODE REVIEW

**Review by boek — APPROVED** [review]
Looks good! Do we also need to show the header when selecting a search engine shortcut?

**Comment by sarah541** [comment]
> Looks good! Do we also need to show the header when selecting a search engine shortcut?

Hey Jeff. Thanks for  the approval! No, we do not show header when selecting shortcut since this is only for default search engine!. Also this PR will be merged once the PR on the AC gets merged : https://github.com/mozilla-mobile/firefox-android/pull/234. Still having a few rounds of review there and I'll add the approach we will be taking in that PR

File: app/src/main/java/org/mozilla/fenix/search/awesomebar/AwesomeBarView.kt [hunk 1]
```diff
@@ -207,6 +210,25 @@ class AwesomeBarView(
         searchSuggestionProviderMap = HashMap()
     }
 
+    private fun getSearchEngineSuggestionsHeader(): String? {
+        val searchState = activity.components.core.store.state.search
+        var searchEngine = searchState.selectedOrDefaultSearchEngine?.name
```
Comment (kycn) [inline app/src/main/java/org/mozilla/fenix/search/awesomebar/AwesomeBarView.kt:215]: nit - maybe injecting components as param to this method and using one variable would be enough as well :)

**Review by kycn — APPROVED** [review]
LGTM! 🎉

**Review by gabrielluong — DISMISSED** [review]
Just wanted to note that we have discussed on Slack yesterday about adding a `enum` in Fenix to have a single source of truth for indicating the provider priority order. This was also independently pointed out by @kycn so I definitely appreciate that this seems to be a common agreed upon pattern. See https://github.com/mozilla-mobile/firefox-android/pull/234/files#r1036865644.

To expand further on the AC side https://github.com/mozilla-mobile/firefox-android/pull/234, the suggestion priorities were hardcoded, which is something we want to avoid because that would dictate all consumer behaviour and goes against the generic component philosophy of AC. Currently, we have a bunch of classes that implements `AwesomeBar.SuggestionProvider` across multiple files that have a specified priority from 1-6. Because everything is spread across multiple file, it’s not very easy to see from a high level what the order of everything is, and that would be a bit sad for developer ergonomics. For example, (1) what happens if someone unfamiliar with the code needs to add a new implementation of `AwesomeBar.SuggestionProvider` and doesn’t understand that priority needs to follow a certain order and doesn’t know we already have 6 other `AwesomeBar.SuggestionProvider` and then randomly puts the `priority = 1` in their implementation or (2) what if in some situations that we don’t show a particular `AwesomeBar.SuggestionProvider` and in those situation we need a different priority order of things because we decided we didn’t want to show the bookmark suggestions. Furthermore, removing a single `AwesomeBar.SuggestionProvider` implementation will mean we need to potentially reorder all of these default values (eg, what happens if somebody removes the class with `priority = 1`, I mean it should be fine, but what if product said the order of everything also had to change along with the removal).

**Pull request merged**